src/AutoRest.CSharp/Common/Output/Models/Requests/RestClientMethod.cs

```diff
@@ -34,5 +36,31 @@ public RestClientMethod(string name, string? description, CSharpType? returnType
         public CSharpType? ReturnType { get; }
         public string Accessibility { get; }
         public Operation? Operation { get; }
+
+        public List<Parameter> NonPathParameters
+        {
+            get
+            {
+                var pathSegments = this.Request.PathParameterSegments;
+
+                return this.Parameters.Where(parameter => !pathSegments.Any(
+                    pathSegment => pathSegment.Value.Reference.Type.Name == parameter.Type.Name &&
+                    pathSegment.Value.Reference.Name == parameter.Name)
+                ).ToList();
+            }
+        }
+
+        public List<Parameter> PathParameters
```

Comment:
Is this used anywhere?

Reply:
My fault, the commit was lost during rebase. I'll add the missing commit soon.